lang/LangSource/PyrInterpreter3.cpp

```diff
@@ -1758,6 +1758,18 @@ HOT void Interpret(VMGlobals *g)
 					PyrSlot * vars = g->frame->vars;
 					int m = slotRawInt(&vars[3]);
 					PyrObject * obj = slotRawObject(&vars[1]);
+					
+					if (IsNil(&vars[1])) {
+						error("primitive failed: keysValuesArrayDo first argument should be an array.\n");
```

Comment:
Can this error message please match the others in this file? See above line 1666:
```
error("Integer-forBy : endval or stepval not an Integer.\n");
```
and below line 1967:
```
error("Number-forSeries : first, second or last not an Integer or Float.\n");
```